File: 9.palindrome-number.rb
#
# @lc app=leetcode id=9 lang=ruby
#
# [9] Palindrome Number
#

# @lc code=start
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
  return false if x.negative?
  str = x.to_s
  return true if str == str.reverse
  false
end
# @lc code=end

File: 9.palindrome-number.rs
/*
 * @lc app=leetcode id=9 lang=rust
 *
 * [9] Palindrome Number
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
      if x < 0 { return false };
      let str: String = x.to_string();
      if str == str.chars().rev().collect::<String>(){ return true };
      return false
    }
}
// @lc code=end
